### PR TITLE
oidc: allow http scheme during discovery 

### PR DESCRIPTION
**Pull request by kegsay**
This is to enable entirely local stacks of Element X to work correctly. It's mostly seamless (no ffi changes) because it ties into `ClientBuilder.insecure_server_name_no_tls`

### CODE REVIEW

**Comment by kegsay** [comment]
Happy with whatever you decide, I just might need pointers on where to touch.

File: crates/matrix-sdk/src/authentication.rs [hunk 1]
```diff
@@ -53,6 +53,10 @@ pub(crate) struct AuthCtx {
     /// `M_UNKNOWN_TOKEN` error is encountered.
     pub(crate) handle_refresh_tokens: bool,
 
+    /// Whether to allow HTTP issuer URLs.
+    #[cfg(feature = "experimental-oidc")]
+    pub(crate) insecure_discover_oidc: bool,
```
Comment (bnjbvr) [inline crates/matrix-sdk/src/authentication.rs:58]: Could you put this in `OidcCtx` instead?

**Review by bnjbvr — CHANGES_REQUESTED** [review]
Almost perfect, thanks!

File: crates/matrix-sdk/src/authentication.rs (hunk 1 above)
Comment (bnjbvr) [reply crates/matrix-sdk/src/authentication.rs]: Done.

**Review by bnjbvr — APPROVED** [review]

**Pull request merged**